orte/util/proc_info.c

```diff
@@ -164,19 +168,29 @@ int orte_proc_info(void)
     /* get the process id */
     orte_process_info.pid = getpid();
 
-    /* get the nodename */
-    gethostname(hostname, sizeof(hostname));
+    /* get the nodename (retrieves from a global variable) */
+    hostname = opal_gethostname();
+    /* copy into a local version so that we can manipulate it */
+    hostname_len = strlen(hostname);
+    if( hostname_len > 0 ) {
+        local_hostname = malloc( strlen(hostname) );
+    }
+    if( NULL == local_hostname ) {
+       return OPAL_ERR_BUFFER; 
+    }
+    strcpy(local_hostname, hostname);
     /* add this to our list of aliases */
-    opal_argv_append_nosize(&orte_process_info.aliases, hostname);
+    opal_argv_append_nosize(&orte_process_info.aliases, local_hostname);
 
     // Strip off the FQDN if present, ignore IP addresses
-    if( !orte_keep_fqdn_hostnames && !opal_net_isaddr(hostname) ) {
-        if (NULL != (ptr = strchr(hostname, '.'))) {
+    if( !orte_keep_fqdn_hostnames && !opal_net_isaddr(local_hostname) ) {
+        if (NULL != (ptr = strchr(local_hostname, '.'))) {
             *ptr = '\0';
             /* add this to our list of aliases */
-            opal_argv_append_nosize(&orte_process_info.aliases, hostname);
+            opal_argv_append_nosize(&orte_process_info.aliases, local_hostname);
         }
     }
+    printf( "local_hostname after strip = %s\n", local_hostname);
```

Comment:
Not sure if this printf was intended to be here or not, but it probably shouldn't be in the final commit.

Reply:
oops. fixed.